### PR TITLE
Clarify the use case for Strict Locals

### PR DESCRIPTION
**Pull request by byroot**
Make it clear that it is to solve the problem of megamorphic partials and that it's not suggested to be used as a default and on all partials.

### CODE REVIEW

File: guides/source/action_view_overview.md [hunk 1]
```diff
@@ -599,22 +599,36 @@ enables compact partial-local default variable assignments:
...
+a partial requires compiling another version:
+
+```html+erb
+<%# app/views/articles/show.html.erb %>
+<%= render partial: "article", layout: "box", locals: { article: @article } %>
+<%= render partial: "article", layout: "box", locals: { article: @article, theme: "dark" } %>
+```
+
+The above snippet will cause the partial to be compiled twice, taking more time and using more memory.
+When the number of combinaisons is small, it's not really a problem, but if it's large, you can use
```
Comment (dallasgoldswain) [inline guides/source/action_view_overview.md:618]: @byroot "combinaisons" is misspelled. It should be: "combinations".

File: guides/source/action_view_overview.md [hunk 2]
```diff
@@ -599,22 +599,36 @@ enables compact partial-local default variable assignments:
...
-more, read about [Strict Locals](#strict-locals).
-
 [local_assigns]:
     https://api.rubyonrails.org/classes/ActionView/Template.html#method-i-local_assigns
 
 ### Strict Locals
 
-Action View partials will accept any number of `locals` as keyword arguments.
+Action View partials are compiled into regular Ruby methods under the hood.
+Because of a Ruby limitation, every single combinaison of `locals` passed to
```
Comment (dallasgoldswain) [inline guides/source/action_view_overview.md:608]: @byroot "combinaison" is misspelled. It should be: "combination".

File: guides/source/action_view_overview.md (hunk 1 above)
Comment (dallasgoldswain) [reply guides/source/action_view_overview.md:618]: commented to help, while learning about the PR

File: guides/source/action_view_overview.md [hunk 3]
```diff
@@ -599,22 +599,36 @@ enables compact partial-local default variable assignments:
...
+a partial requires compiling another version:
```
Comment (jhawthorn) [inline guides/source/action_view_overview.md:609]: I don't mind this wording, but don't really see it as a Ruby limitation so much as a quirk of Ruby's syntax

```suggestion
In Ruby method calls and local variable access look the same, so every single
combination of `locals` passed to a partial requires compiling another version:
```

If you don't think that's clearer though, yours is fine. 👍

**Review by jhawthorn — APPROVED** [review]

**Comment by gregmolnar** [comment]
@byroot This change should be reflected on the edgeguides, right? I still see the old version, is it just me? 
https://edgeguides.rubyonrails.org/action_view_overview.html#strict-locals
image

**Comment by byroot** [comment]
Yeah, not too sure why edge guide are lagging behind, I don't know where they are published.

**Pull request merged**